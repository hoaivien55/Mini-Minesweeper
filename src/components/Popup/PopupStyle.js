import styled from 'styled-components';


const PopupStyle = styled.div`
width: 100vw;
height: 100vh;
position: fixed;
left: 0;
top: 0;
background: #76c3c34d;
z-index: 100;
`;

const PopupContainerStyle = styled.div`
width: 100%;
height: 100%;
max-width: ${(props) => props.maxWidth || "400px"};
max-height: ${(props) => props.maxHeight || "200px"};
min-width: 200px;
min-height: 200px;
margin: 30px auto;
background: white;
border-radius: 12px;
`;

const PopupBodyStyle = styled.div`
width: 100%;
height: ${(props) => props.height || "70%"};
display: flex;
justify-content: center;
align-items: center;
.paragraph {
    font-size: 25px;
}
`;

const PopupFooter = styled.div`
width: 100%;
height: ${(props) => props.height || "30%"};
display: flex;
align-items: center;
justify-content: center;
border-top: 1px solid #80808030;
button:first-child {
    margin-right: 15px;
}
`;

export {
    PopupStyle,
    PopupContainerStyle,
    PopupBodyStyle,
    PopupFooter
}