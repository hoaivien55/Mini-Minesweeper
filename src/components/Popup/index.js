import PropTypes from "prop-types";
import { memo } from "react";

import { Button, Row } from "../styles";
import {
	PopupStyle,
	PopupContainerStyle,
	PopupBodyStyle,
	PopupFooter,
} from "./PopupStyle";

const Popup = ({ accept, reject, acceptLabel, rejectLabel, message }) => {
	return (
		<PopupStyle>
			<PopupContainerStyle>
				<PopupBodyStyle>
					<Row className="paragraph">{message}</Row>
				</PopupBodyStyle>
				<PopupFooter>
					<Button onClick={accept}>{acceptLabel}</Button>
					<Button onClick={reject}>{rejectLabel}</Button>
				</PopupFooter>
			</PopupContainerStyle>
		</PopupStyle>
	);
};

Popup.defaultProps = {
	accept: () => {},
	reject: () => {},
	acceptLabel: "Accept",
	rejectLabel: "Reject",
};

Popup.propTypes = {
	accept: PropTypes.func,
	reject: PropTypes.func,
	acceptLabel: PropTypes.string,
	rejectLabel: PropTypes.string,
	message: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
};

export default memo(Popup);
