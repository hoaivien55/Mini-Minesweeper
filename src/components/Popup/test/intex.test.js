import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";

import Popup from "..";

it("Popup: Virtual DOM snapshot", () => {
	const tree = renderer.create(<Popup message={"hello there"} />).toJSON();
	expect(tree).toMatchSnapshot();
});

test("Popup: change message", async () => {
	const message = "hello Adam";
	render(<Popup message={message} />);
	expect(screen.getByText(message)).toBeInTheDocument();
});

test("Popup: accept button click", async () => {
	const message = "hello Adam";
	const acceptClick = jest.fn();
	render(<Popup message={message} accept={acceptClick} />);
	const button = screen.getByText(/Accept/);
	fireEvent.click(button);
	expect(acceptClick).toHaveBeenCalledTimes(1);
});

test("Popup: reject button click", async () => {
	const message = "hello Adam";
	const rejectClick = jest.fn();
	render(<Popup message={message} reject={rejectClick} />);
	const button = screen.getByText(/Reject/);
	fireEvent.click(button);
	expect(rejectClick).toHaveBeenCalledTimes(1);
});

test("Popup: button label", async () => {
	const message = "hello Adam";
	const acceptLabel = "test 1";
	const rejecttLabel = "test 2";
	render(
		<Popup
			message={message}
			acceptLabel={acceptLabel}
			rejectLabel={rejecttLabel}
		/>
	);
	expect(screen.getByText(new RegExp(acceptLabel))).toBeInTheDocument();
	expect(screen.getByText(new RegExp(rejecttLabel))).toBeInTheDocument();
});
