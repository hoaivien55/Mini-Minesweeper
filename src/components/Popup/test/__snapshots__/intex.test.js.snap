// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup: Virtual DOM snapshot 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px;
}

.c5 {
  min-width: 120px;
  background: #65dfa1;
  border: none;
  height: 30px;
  border-radius: 12px;
  color: white;
  font-weight: bold;
  text-transform: uppercase;
  cursor: pointer;
}

.c5:hover {
  opacity: 0.8;
}

.c0 {
  width: 100vw;
  height: 100vh;
  position: fixed;
  left: 0;
  top: 0;
  background: #76c3c34d;
  z-index: 100;
}

.c1 {
  width: 100%;
  height: 100%;
  max-width: 400px;
  max-height: 200px;
  min-width: 200px;
  min-height: 200px;
  margin: 30px auto;
  background: white;
  border-radius: 12px;
}

.c2 {
  width: 100%;
  height: 70%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .paragraph {
  font-size: 25px;
}

.c4 {
  width: 100%;
  height: 30%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-top: 1px solid #80808030;
}

.c4 button:first-child {
  margin-right: 15px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3 paragraph"
      >
        hello there
      </div>
    </div>
    <div
      className="c4"
    >
      <button
        className="c5"
        onClick={[Function]}
      >
        Accept
      </button>
      <button
        className="c5"
        onClick={[Function]}
      >
        Reject
      </button>
    </div>
  </div>
</div>
`;
